src/autocompleteDump.ts

```diff
@@ -0,0 +1,124 @@
+import * as request from "request-promise-native"
+import * as vscode from "vscode"
+import { parseStringPromise } from "xml2js"
+
+const AUTOCOMPLETE_METADATA = "https://raw.githubusercontent.com/CloneTrooper1019/Roblox-Client-Tracker/roblox/AutocompleteMetadata.xml"
+
+export interface AutocompleteParameter {
+    type: string,
+    name: string,
+    optional: boolean
+}
+
+export interface AutocompleteReturn {
+    type: string,
+    name: string
+}
+
+export interface AutocompleteFunction {
+    name: string,
+    static: boolean,
+    description?: string,
+    parameters: AutocompleteParameter[]
+    returns: AutocompleteReturn[]
+}
+
+export interface AutocompleteProperty {
+    name: string,
+    static: boolean,
+    type: string,
+    description?: string
+}
+
+export interface AutocompleteGroup {
+    name: string,
+    functions: AutocompleteFunction[],
+    properties: AutocompleteProperty[]
+}
+
+export interface AutocompleteDump {
+    LuaLibrary: AutocompleteGroup[],
+    ItemStruct: AutocompleteGroup[]
+}
+
+const formatFunction = (func: any): AutocompleteFunction => {
+    const attributes = func.$
+    const parameters: AutocompleteParameter[] = 
+        (func.parameters && func.parameters[0].$$) ? func.parameters[0].$$.map((paramObj: any) => {
```

Comment:
Is `$$` a Roblox XML thing? That's awful.

Reply:
Also don't use `any`

Reply:
The $$ is injected by the xml2js parser. It really isn’t that nice, but I’ve had to set it up this way in order to keep the children order, as that is very important for parameters. I tried using fast-xml-parser instead, but it didn’t preserve order (from first glance)